pkgs/development/tools/backblaze-b2/default.nix

```diff
@@ -21,6 +22,8 @@ python3Packages.buildPythonApplication rec {
       --replace 'tabulate==0.8.10' 'tabulate'
     substituteInPlace setup.py \
       --replace 'setuptools_scm<6.0' 'setuptools_scm'
+    substituteInPlace b2/console_tool.py \
+      --replace 'import sys' 'import sys; sys.argv[0]="${execName}"'
```

Comment:
I'm not a huge fan of patching the source like this. What if they start using another library that starts with `sys`?

If the benefit is just improving `--help` it doesn't seem worth the risk/cost. 

My thoughts would be:

- Can we change how we exec from the wrapper to set `sys.arvg[0]` "correctly" before the program starts?
- Can we send a patch upstream to just show the basename in help?
- Can we carry our own real patch that at least matches the whole line so is less likely to break other things?

Reply:
Thanks. I've found a better way of doing this and submitted a PR upstream: https://github.com/Backblaze/B2_Command_Line_Tool/pull/993.

Should we take this patch into nixpkgs?

Reply:
Thanks, that looks like a much better solution.

I don't have any objections to carrying this patch in nixpkgs for a while.